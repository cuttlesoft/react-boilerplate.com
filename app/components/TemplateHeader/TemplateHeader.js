import React from 'react'
import { FormattedMessage } from 'react-intl'

import { Anchor } from '../Anchor'
import { Image } from '../Image'
import { Box } from '../Box'
import NavBar from './NavBar'
import HeaderLink from './HeaderLink'
import Banner from '../../images/icon-512x512.png'
import messages from './messages'

function TemplateHeader() {
  return (
    <div>
      <Box align="center" gap="medium" height="medium">
        <Anchor href="https://www.reactboilerplate.com/">
          <Box height="medium" width="medium">
            <Image fit="contain" src={Banner} alt="react-boilerplate - Logo" />
          </Box>
        </Anchor>
      </Box>
      <NavBar>
        <HeaderLink to="/">
          <FormattedMessage {...messages.home} />
        </HeaderLink>
        <HeaderLink to="/features">
          <FormattedMessage {...messages.features} />
        </HeaderLink>
      </NavBar>
    </div>
  )
}

export default TemplateHeader
