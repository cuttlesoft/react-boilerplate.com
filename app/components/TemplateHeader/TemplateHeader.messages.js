import { defineMessages } from 'react-intl'

export const scope = 'boilerplate.components.Header'

/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
export default defineMessages({
  home: {
    id: `${scope}.home`,
    defaultMessage: 'Home',
  },
  features: {
    id: `${scope}.features`,
    defaultMessage: 'Features',
  },
})
