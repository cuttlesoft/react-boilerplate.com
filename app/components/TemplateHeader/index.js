export { default as TemplateHeader } from './TemplateHeader'
