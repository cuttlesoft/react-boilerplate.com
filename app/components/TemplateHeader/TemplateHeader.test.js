/**
 *
 * Tests for TemplateHeader
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

import 'jest-styled-components'

import TemplateHeader from './TemplateHeader'

const renderComponent = (props = {}) =>
  render(
    <BrowserRouter>
      <TemplateHeader {...props} />)
    </BrowserRouter>,
  )

describe('TemplateHeader', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })
})
