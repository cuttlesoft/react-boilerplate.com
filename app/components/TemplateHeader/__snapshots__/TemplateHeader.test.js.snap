// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateHeader renders and matches snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 384px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 384px;
  width: 384px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  overflow: hidden;
  object-fit: contain;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0.25em 2em;
  margin: 1em;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: bold;
  font-size: 16px;
  border: 2px solid #41addd;
  color: #41addd;
}

.c5:active {
  background: #41addd;
  color: #fff;
}

<div>
  <div>
    <div
      class="c0"
    >
      <a
        class="c1"
        href="https://www.reactboilerplate.com/"
      >
        <div
          class="c2"
        >
          <img
            alt="react-boilerplate - Logo"
            class="c3"
            src="IMAGE_MOCK"
          />
        </div>
      </a>
    </div>
    <div
      class="c4"
    >
      <a
        class="c5"
        href="/"
      >
        <span>
          Home
        </span>
      </a>
      <a
        class="c5"
        href="/features"
      >
        <span>
          Features
        </span>
      </a>
    </div>
  </div>
  )
</div>
`;
